lucene/core/src/java/org/apache/lucene/search/ConjunctionDISI.java

```diff
@@ -140,6 +141,13 @@ private static void addTwoPhaseIterator(TwoPhaseIterator twoPhaseIter, List<DocI
   private static DocIdSetIterator createConjunction(
       List<DocIdSetIterator> allIterators,
       List<TwoPhaseIterator> twoPhaseIterators) {
+
+    // assert that all sub-iterators are on the same doc ID
```

Comment:
@jpountz  Is this a correct assumption?  or could we start with iterators that are not on the same document?

Reply:
this is the invariant between iterations indeed